osh/word_eval.py

```diff
@@ -887,19 +887,22 @@ def _EvalVarRef(self, val, blame_tok, quoted, vsub_state, vtest_place):
 
             elif case(value_e.Str):
                 val = cast(value.Str, UP_val)
-                bvs_part = self.unsafe_arith.ParseVarRef(val.s, blame_tok)
-                return self._VarRefValue(bvs_part, quoted, vsub_state,
-                                         vtest_place)
+                var = val.s
 
             elif case(value_e.BashArray):  # caught earlier but OK
-                e_die('Indirect expansion of array')
+                val = cast(value.BashArray, UP_val)
+                var = ' '.join(bash_impl.BashArray_GetValues(val))
```

Comment:
So this creates an INVALID bvs_str whenever the array or assoc has more than one value?  

Because something like `${a b c}` is never valid in bash/OSH

That part was confusing me.  I think that deserves a comment here

---

Actually a funny thing is this may be valid in YSH - I want to use the syntax

```
var y = ${x %2.f}
```

to replace `printf -v y '%.2f' $x`

Reply:
#2204